android/src/main/java/com/oney/WebRTCModule/PeerConnectionObserver.java

```diff
@@ -455,6 +369,125 @@ public void onSignalingChange(PeerConnection.SignalingState signalingState) {
     @Override
     public void onAddTrack(final RtpReceiver receiver, final MediaStream[] mediaStreams) {
         Log.d(TAG, "onAddTrack");
+
+        // Get The timestamp so that we can reorder the transceivers on the JS layer
```

Comment:
Yes but if you keep the JS and native in sync then they have matching indexes in the transceivers array don't they?

Reply:
Oh I don't think that was the intention. The idea is if addTransceiver(...) returns a transceiver object, that exact object has to be the one receiving updates. IMO it would be better to add transceivers one by one. Otherwise, if we serialize the whole transceivers array, the only thing we would gain from it is their order. I think adding one timestamp to the returned object is probably more efficient than a whole array of transceivers, don't you think?

Reply:
Gotcha. In that case how about this: each PC will start with a transceiver counter of 0. Every time a new transceiver is crated we use it and increment it. I think the code will be more concise and easier to follow like that.